### ISSUE TITLE
Dynamic update of the Select type's titleMap not working for the array type schema which has an select property.

### ISSUE THREAD

**Issue by ravikumarmistry** [json-schema-form/angular-schema-form#823]
#### Enhancement
As a user/developer, when I try to update "select" type's titleMap it is not updating if it is in Array.

I have illustrated it in the [Demo](http://jsbin.com/tirekud/1/edit?js,output). 

#### Gist/Plunker/Demo
[Demo](http://jsbin.com/tirekud/1/edit?js,output)

#### Related issues
This is/maybe related to copy of titleMap object and passed to the Array decorator instead of referencing the titleMap for Array decorator.  

@json-schema-form/angular-schema-form-lead

**Comment by Anthropic** [comment]
@ravikumarmistry the titleMap isn't meant to be a watched value, that is a side effect of angular integration. There are a few add-ons that can allow for watching an object to populate a titleMap, but it would be useful and I will leave this as an enhancement as it would be nice if the solution in the material decorator worked in bootstrap also.

**Comment by ravikumarmistry** [comment]
@Anthropic is this implemented in the material decorator.

**Comment by Anthropic** [comment]
Yes, although the Material decorator has other issues... the add-ons would be your best option for now I expect.

**Comment by ravikumarmistry** [comment]
@Anthropic i am thinking of creating an add on, I want to expose some of my controllers $scope property to be exposed in the form decorator. how can i archive it. Any guidance would be helpful.

**Comment by Anthropic** [comment]
You could add an angular directive to the template you make for the add-on that passes a value into the scope of the current field.
